python/python_helper_sympy.py

```diff
@@ -264,98 +323,105 @@ def evaluate(expr, locals_for_eval={}):
-                **locals_for_eval["variables"],
-                **const.hidden_complex_variables,
-            }
+            locals_for_eval["variables"].update(const.hidden_complex_variables)
+
+    if allow_trig_functions:
+        locals_for_eval["functions"].update(const.trig_functions)
 
     # If there is a list of variables, add each one to the whitelist
     if variables is not None:
-        for variable in variables:
-            locals_for_eval["variables"][variable] = sympy.Symbol(variable)
+        locals_for_eval["variables"].update(
+            (variable, sympy.Symbol(variable)) for variable in variables
+        )
 
     # Do the conversion
-    return evaluate(a, locals_for_eval)
+    return evaluate(expr, locals_for_eval)
 
 
-def point_to_error(s, ind, w=5):
-    w_left = ind - max(0, ind - w)
-    w_right = min(ind + w, len(s)) - ind
-    return s[ind - w_left : ind + w_right] + "\n" + " " * w_left + "^" + " " * w_right
+def point_to_error(expr: str, ind: int, w: int = 5) -> str:
+    """Generate a string with a pointer to error in expr with index ind"""
+    w_left: str = " " * (ind - max(0, ind - w))
+    w_right: str = " " * (min(ind + w, len(expr)) - ind)
+    initial: str = expr[ind - len(w_left) : ind + len(w_right)]
+    return f"{initial}\n{w_left}^{w_right}"
 
 
-def sympy_to_json(a, allow_complex=True):
+def sympy_to_json(
+    a: sympy.Expr, *, allow_complex: bool = True, allow_trig_functions: bool = True
+) -> SympyJson:
     const = _Constants()
 
     # Get list of variables in the sympy expression
-    variables = [str(v) for v in a.free_symbols]
+    variables = list(map(str, a.free_symbols))
 
     # Check that variables do not conflict with reserved names
-    reserved = {
-        **const.helpers,
-        **const.variables,
-        **const.hidden_variables,
-        **const.functions,
-    }
+    reserved = (
+        const.helpers.keys()
+        | const.variables.keys()
+        | const.hidden_variables.keys()
+        | const.functions.keys()
+    )
     if allow_complex:
-        reserved = {
-            **reserved,
-            **const.complex_variables,
-            **const.hidden_complex_variables,
-        }
-    for k in reserved.keys():
-        for v in variables:
-            if k == v:
-                raise ValueError(
-                    "sympy expression has a variable with a reserved name: {:s}".format(
-                        k
-                    )
-                )
+        reserved |= (
+            const.complex_variables.keys() | const.hidden_complex_variables.keys()
+        )
+    if allow_trig_functions:
+        reserved |= const.trig_functions.keys()
+
+    # Check if reserved variables conflict, raise an error if they do
+    conflicting_reserved_variables = reserved & set(variables)
+
+    if conflicting_reserved_variables:
+        raise ValueError(
+            f"sympy expression has a variable with a reserved name: {conflicting_reserved_variables}"
```

Comment:
Nit: reflecting the fact that there may be multiple conflicts.

```suggestion
            f"sympy expression has variables with reserved names: {conflicting_reserved_variables}"
```